fedimint-client/src/sm/executor.rs

```diff
@@ -504,29 +512,31 @@ impl ExecutorInner {
                         warn!(target: LOG_CLIENT_REACTOR, operation_id = %state.operation_id().fmt_short(), "Received a state machine that is already running. Ignoring");
                         continue;
                     }
-                    let Some(meta) = self.get_active_state(&state).await else {
-                        warn!(target: LOG_CLIENT_REACTOR, operation_id = %state.operation_id().fmt_short(), "Couldn't look up received state machine. Ignoring.");
-                        continue;
-                    };
+                    currently_running_sms.insert(state.clone());
+                    let futures_len = futures.len();
+                    let global_context_gen = &global_context_gen;
+                    trace!(target: LOG_CLIENT_REACTOR, state = ?state, "Started new active state machine, details.");
+                    futures.push(Box::pin(async move {
+                        let Some(meta) = self.get_active_state(&state).await else {
```

Comment:
This and and below check require `.await` during which other state machines are not being polled. It doesn't seem like much, but it is relatively easy to make them inside a new future, and if the checks fail, return a new type of a loop event. This way the main executor loop only ever polls all `futures`, and adds new futures to `futures`, avoiding any problems with `FuturesUnordered`.

Reply:
What's reason to not spawn a task for each contained future here? Performance?